File: gh.js
const axios = require('axios');
const readline = require('readline');

let ClrPrinted = false;

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const color = {
    ul: '\x1b[4m',
    green: '\x1b[32m',
    red: '\x1b[31m',
    blue: '\x1b[34m',
    yellow: '\x1b[33m',
    reset: '\x1b[0m'
};

function clr() {
    if (!ClrPrinted) {
        console.log(`
╭━━━╮╱╱╱╱╱╱╱╱╱╱╱╱╱╭━━━┳╮
┃╭━━╯╱╱╱╱╱╱╱╱╱╱╱╱╱┃╭━━┫┃${color.blue}Github${color.green}
┃╰━━┳╮╭┳━┳━━┳━━┳━╮┃╰━━┫┃╭╮╱╭┳━╮╭━╮
┃╭━━┫┃┃┃╭┫╭╮┃╭╮┃╭╮┫╭━━┫┃┃┃╱┃┃╭╮┫╭╮╮
┃┃╱╱┃╰╯┃┃┃╰╯┃╰╯┃┃┃┃┃╱╱┃╰┫╰━╯┃┃┃┃┃┃┃
╰╯╱╱╰━━┻╯╰━╮┣━━┻╯╰┻╯╱╱╰━┻━╮╭┻╯╰┻╯╰╯
╱╱╱╱╱╱╱╱╱╱╱┃┃╱╱╱╱╱╱╱╱╱╱╱╭━╯┃${color.reset}Username
╱╱╱╱╱╱╱╱╱╱╱╰╯╱╱╱╱╱╱╱╱╱╱╱╰━━╯${color.yellow}Checker
\n[+]${color.green} Type "cl" to clear the screen${color.reset}
        `);
        ClrPrinted = true;
    }
}

async function checkUsername(username) {
    try {
        // https://api.github.com/users/ (has limitations)
        const response = await axios.get(`https://github.com/${username}`, {
            headers: {
                'User-Agent': 'Node.js'
            }
        });
        console.log(color.red + `Username "${username}" is already taken.` + color.reset);
        askForUsername();
    } catch (error) {
        if (error.response && error.response.status === 404) {
            console.log(color.green + `Username "${username}" is available.` + color.reset);
            rl.question('Enter another username or type "exit" to quit: ', (newUsername) => {
                if (newUsername.toLowerCase() === 'exit') {
                    rl.close();
                } else {
                    checkUsername(newUsername);
                }
            });
        } else {
            console.log(color.yellow + `Error: ${error.message}` + color.reset);
            askForUsername();
        }
    }
}

function askForUsername() {
    clr();
    rl.question('Enter a GitHub username to check: ', (input) => {
        if (input.toLowerCase() === 'cl') {
            console.clear();
            askForUsername();
        } else {
            checkUsername(input);
        }
    });
}

askForUsername();

File: gh.rb
require 'httparty'

Signal.trap("INT") do
  puts "\nExiting gracefully..."
  exit
end

def clr
  puts "
╭━━━╮╱╱╱╱╱╱╱╱╱╱╱╱╱╭━━━┳╮
┃╭━━╯╱╱╱╱╱╱╱╱╱╱╱╱╱┃╭━━┫┃\e[34mGithub\e[32m
┃╰━━┳╮╭┳━┳━━┳━━┳━╮┃╰━━┫┃╭╮╱╭┳━╮╭━╮
┃╭━━┫┃┃┃╭┫╭╮┃╭╮┃╭╮┫╭━━┫┃┃┃╱┃┃╭╮┫╭╮╮
┃┃╱╱┃╰╯┃┃┃╰╯┃╰╯┃┃┃┃┃╱╱┃╰┫╰━╯┃┃┃┃┃┃┃
╰╯╱╱╰━━┻╯╰━╮┣━━┻╯╰┻╯╱╱╰━┻━╮╭┻╯╰┻╯╰╯
╱╱╱╱╱╱╱╱╱╱╱┃┃╱╱╱╱╱╱╱╱╱╱╱╭━╯┃\e[0mUsername
╱╱╱╱╱╱╱╱╱╱╱╰╯╱╱╱╱╱╱╱╱╱╱╱╰━━╯\e[33mChecker

\e[32m[+] Type \"cl\" to clear the screen\e[0m
  "
end

def check_username(username)
  response = HTTParty.get("https://github.com/#{username}", headers: { 'User-Agent' => 'Ruby' })
  if response.code == 404
    puts "\e[32mUsername \"#{username}\" is available.\e[0m"
    ask_for_another_username
  else
    puts "\e[31mUsername \"#{username}\" is already taken.\e[0m"
    ask_for_username
  end
rescue HTTParty::ResponseError => e
  puts "Error: #{e.message}"
  ask_for_username
end

def ask_for_another_username
  print 'Enter another username or type "exit" to quit: '
  input = gets.chomp
  if input.downcase == 'exit'
    puts "Exiting..."
    exit
  else
    check_username(input)
  end
end

def ask_for_username
  print 'Enter a GitHub username to check: '
  input = gets.chomp
  if input.downcase == 'cl'
    system('clear') || system('cls')
    ask_for_username
  else
    check_username(input)
  end
end

clr
ask_for_username

